plutus-core-interpreter/src/Language/PlutusCore/Interpreter/LMachine.hs

```diff
@@ -0,0 +1,278 @@
+-- | The L machine
+-- A lazy machine based on the L machine of Friedman et al. [Improving the Lazy Krivine Machine]
+-- For more details see the document in plutus/docs/fomega/lazy-machine
+-- The code here's closely based on the CEK machine implementation.
+
+-- The main difference from the CEK machine is that we have a heap
+-- containing closures containing evaluated/unevaluated function
+-- arguments, and environments mapping variable IDs to heap locations.
+-- To evaluate [M N], we save a closure C containing N and the current
```

Comment:
I'm not sure this is going to look good in Haddock...

Reply:
And throughout.

Reply:
Personally, I'm trying to follow haddock's silliness, but I'm pretty sure I've failed multiple times to do so. We need someone who can actually check all the generated haddocks and fix where appropriate.